syntax = "proto3";

package telemetry.v1;

import "google/protobuf/timestamp.proto";

service TelemetryService {
  rpc StreamTelemetry (stream TelemetryRequest) returns (stream TelemetryResponse);
}

message TelemetryRequest {
  string sensor_name = 1;
  int64 sensor_value = 2;
  google.protobuf.Timestamp timestamp = 3;
}

message TelemetryResponse {
  enum Status {
    STATUS_UNSPECIFIED = 0;
    OK = 1;
    RATE_LIMITED = 2;
    ERROR = 3;
  }
  Status status = 2;
}